src/nibpkg/compose.nim

```diff
@@ -6,15 +6,35 @@ type
     left, right: string
 
 proc retrieve_flanking_sequences_from_fai(fastaIdx: Fai, chrom: string,
-        start_pos : int, end_pos : int, flank: int): FlankSeq =
+        start_pos: int, end_pos: int, flank: int): FlankSeq =
   ## this function lacks a return
   result.left = fastaIdx.get(chrom, max(0, start_pos - flank), start_pos)
   result.right = fastaIdx.get(chrom, end_pos, end_pos + flank)
 
+proc compose(variant: Variant, right_flank: string,
+    left_flank: string): string =
 
+  var inner_seq: string
+  var variant_type: string
+  doAssert variant.info.get("SVTYPE", variant_type) == Status.OK
+  if variant_type == "DEL":
+    inner_seq = ""
+  elif variant_type == "INS":
+    inner_seq = variant.ALT[0]
+  elif variant_type == "INV":
+    raise newException(ValueError,
+    "Error: Inversion processing not implemented.")
+  var combined_sequence = right_flank & inner_seq & left_flank
 
-proc compose_variants*(variant_file: string, reference_file: string) =
-  ## function to compose
+  return combined_sequence
+
+
+proc compose_variants*(variant_file: string, reference_file: string): seq[string] =
+  ## function to compose variants from their sequence / FASTA flanking regions
+  ## Returns a Sequence of strings representing the DNA sequence of the flanking
+  ## regions and variant sequence.
+
+  var composed_seqs = newSeq[string](1)
```

Comment:
`newSeq[string]()`

It can start out empty. The `.add()` function extends the array by 1.